.github/workflows/cicd.yaml

```diff
@@ -70,33 +72,40 @@ jobs:
         paths.src_las=/inputs/888000_6614000.subset.las
         paths.output_dir=/outputs/
 
-    # Everything ran so we tag the valid docker image to keep it
-    # This happens for push events, which are in particular 
-    # triggered when a pull request is merged.
-    - name: Get the branch name into an environment variable.
-      if: github.event_name == 'push'
-      uses: nelonoel/branch-name@v1.0.1
-
-    - name: Print the branch name
-      if: github.event_name == 'push'
-      run: echo ${BRANCH_NAME}
-
     - name: Tag the docker image with branch name
       if: github.event_name == 'push'
-      run: docker tag lidar_prod_im:latest lidar_prod_im:${BRANCH_NAME}
-
-    - name: Dry run image so that is it not prunned
-      if: github.event_name == 'push'
-      run: docker run lidar_prod_im:${BRANCH_NAME} bash
-
-    # This needs writing rights to the mounted path
-    - name: save the docker image because everything worked
-      if: github.event_name == 'push'
-      run: docker save lidar_prod_im:${BRANCH_NAME} -o /var/data/cicd/CICD_github_assets/CICD_docker_images/lidar_prod_im_${BRANCH_NAME}.tar
+      run: |
+        docker tag lidar_prod_im:latest lidar_prod_im:${{github.ref_name}} 
+        docker run lidar_prod_im:${{github.ref_name}}  bash  # Dry run image so that is it not prunned
+        # docker save lidar_prod_im:${{github.ref_name}}  -o /var/data/cicd/CICD_github_assets/CICD_docker_images/lidar_prod_im_${{github.ref_name}}.tar # This needs writing rights to the mounted path
+
+    # get version number and date, to tag the image pushed to nexus
+    - name: Get version number
+      id: tag
+      run: |
+        echo "::set-output name=version::$(docker run myria3d grep '__version__' package_metadata.yaml| cut -d\" -f2)"
+        echo "::set-output name=date::$(date '+%Y.%m.%d')"
+
+    # show possible tags, for debugging purpose
+    - name: Print tag
+      run: |
+        echo "${{steps.tag.outputs.version}}"
+        echo "${{steps.tag.outputs.date}}"
+
+    - name: Push docker image on Nexus repository if a pull request on main has been accepted.  
+      if: ((github.ref_name == 'main') && (github.event.pull_request.merged == true)) # we push on nexus only if a pull request on the main branch has been accepted 
+      run: |
+        docker tag lidar_prod_im $nexus_server/lidar_hd/lidar_prod_im:${{steps.tag.outputs.date}}
+        docker login $nexus_server --username svc_lidarhd --password ${{ secrets.PASSWORD_SVC_LIDARHD }}
+        docker push $nexus_server/lidar_hd/lidar_prod_im:${{steps.tag.outputs.date}}
+
+    - name: Push docker image on nexus repository, when pushing to any other branch.
+      if: ((github.event_name != 'pull_request') && (github.ref_name != 'main'))
```

Comment:
Est-ce qu'on peut écrire de manière explicite que c'est si event_name=push qu'on fait cette action ? Ce sera plus simple à lire.

Reply:
Oui, j'ai remplacé par event_name==push, je ne sais pas pourquoi je suis passé par l'inégalité